src/commonMain/kotlin/exchange.dydx.abacus/calculator/AdjustIsolatedMarginInputCalculator.kt

```diff
@@ -0,0 +1,140 @@
+
+        return if (wallet != null && isolatedMarginAdjustment != null && type != null) {
+            val modified = state.mutable()
+            val parentTransferDelta = getModifiedTransferDelta(isolatedMarginAdjustment, true)
+            val childTransferDelta = getModifiedTransferDelta(isolatedMarginAdjustment, false)
+
+            val walletPostParentSubaccountTransfer =
+                subaccountTransformer.applyIsolatedMarginAdjustmentToWallet(
+                    wallet,
+                    subaccountNumber = parentSubaccountNumber,
+                    parentTransferDelta,
+                    parser,
+                    "postOrder",
+                )
+
+            val walletPostChildSubaccountTransfer =
+                subaccountTransformer.applyIsolatedMarginAdjustmentToWallet(
+                    wallet = walletPostParentSubaccountTransfer,
+                    subaccountNumber = childSubaccountNumber,
+                    childTransferDelta,
+                    parser,
+                    "postOrder",
+                )
+
+            val modifiedParentSubaccount = parser.asNativeMap(parser.value(walletPostChildSubaccountTransfer, "accounts.subaccounts.$parentSubaccountNumber"))
+            val modifiedChildSubaccount = parser.asNativeMap(parser.value(walletPostChildSubaccountTransfer, "accounts.subaccounts.$childSubaccountNumber"))
+            val modifiedIsolatedMarginAdjustment = finalize(isolatedMarginAdjustment, modifiedParentSubaccount, modifiedChildSubaccount, type)
+
+            modified["adjustIsolatedMargin"] = modifiedIsolatedMarginAdjustment
+            modified["wallet"] = walletPostChildSubaccountTransfer
+            modified
+        } else {
+            state
+        }
+    }
+
+    private fun getModifiedTransferDelta(
+        isolatedMarginAdjustment: Map<String, Any>,
+        isParentSubaccount: Boolean,
+    ): Map<String, Double>? {
+        val type = parser.asString(isolatedMarginAdjustment["type"])
+        val amount = parser.asDouble(isolatedMarginAdjustment["amount"])
+        if (type != null) {
+            when (type) {
+                "ADD" -> {
+                    val multiplier =
+                        if (isParentSubaccount) Numeric.double.NEGATIVE else Numeric.double.POSITIVE
+                    val usdcSize = (amount ?: Numeric.double.ZERO) * multiplier
+
+                    return mapOf(
+                        "usdcSize" to usdcSize,
+                    )
+                }
+
+                "REMOVE" -> {
+                    val multiplier =
+                        if (isParentSubaccount) Numeric.double.POSITIVE else Numeric.double.NEGATIVE
+                    val usdcSize = (amount ?: Numeric.double.ZERO) * multiplier
+
+                    return mapOf(
+                        "usdcSize" to usdcSize,
+                    )
+                }
+
+                else -> {
+                    return null
```

Comment:
is this a valid state to be in? if not, i'd recommend just `error()`ing out here.

Reply:
or even better, we could parse this into an enum and get exhaustive `when` checking here.

Reply:
good call, parsed to string but invoked via enum class.

Reply:
hmm i'm not seeing an enum used here?